index.go

```diff
@@ -268,45 +273,49 @@ func (i *index) Iterate(bucket *bbolt.Bucket, query Query, fn IteratorFn) error
 		return err
 	}
 
-	return findR(cBucket.Cursor(), Key{}, sortedQueryParts, fn)
+	return findR(cBucket.Cursor(), Key{}, sortedQueryParts, i.indexParts, fn)
 }
 
-func findR(cursor *bbolt.Cursor, sKey Key, parts []QueryPart, fn IteratorFn) error {
+func findR(cursor *bbolt.Cursor, sKey Key, parts []QueryPart, indexParts []IndexPart, fn IteratorFn) error {
 	cPart := parts[0]
 	seek, err := cPart.Seek()
 	if err != nil {
 		return err
 	}
-
-	seek = ComposeKey(sKey, seek)
-	condition := true
-	for cKey, entry := cursor.Seek(seek); cKey != nil && bytes.HasPrefix(cKey, sKey) && condition; cKey, entry = cursor.Next() {
-		// remove prefix (+1), Split and take first
-		pf := cKey[len(sKey)+1:]
-		if len(sKey) == 0 {
-			pf = cKey
-		}
-		pfk := Key(pf)
-		newp := pfk.Split()[0] // todo bounds check?
-
-		condition, err = cPart.Condition(newp)
-		if err != nil {
-			return err
-		}
-		if condition {
-			if len(parts) > 1 {
-				nKey := ComposeKey(sKey, newp)
-				err = findR(cursor, nKey, parts[1:], fn)
-
-			} else {
-				err = fn(cKey, entry)
+	// value/search terms transformation (eg lowercase)
+	tokens := indexParts[0].Tokenize(seek)
```

Comment:
quite complex, could use some refactoring maybe?

Reply:
It does require a refactor, but this PR is not the time to do it.